### DISCUSSION TITLE
I cannot refer to the stored value

### DISCUSSION THREAD

**Discussion by martynaol** [pactumjs/pactum#130] [Q&A]
I want to store the access token to use this value later. I have : 

```js
const pactum = require('pactum');
const { Given, When, Then, Before } = require('@cucumber/cucumber');

Before(async () => {
    await pactum
      .spec()
      .post('/credentials')
      .withForm({
        username: email,
        password: passw,
      })
      .expectStatus(200)
      .stores('token', 'access_token');
})
    pactum.request.setDefaultHeaders('Authorization', 'Bearer $S{token}');
```
syntax '$S{token}' is a string, the next request is with   "headers": {
    "Authorization": "Bearer $S{token}"
  },

what am I doing wrong?

**Comment by ASaiAnudeep** [comment]
Try invoking the `setDefaultHeaders` method inside **Before** block after storing the token.

```js
const { spec, request } = require('pactum');
const { Given, When, Then, Before } = require('@cucumber/cucumber');

Before(async () => {
  await spec()
      .post('/credentials')
      .withForm({
        username: email,
        password: passw,
      })
      .expectStatus(200)
      .stores('token', 'access_token');
  request.setDefaultHeaders('Authorization', 'Bearer $S{token}');
});
```

**Reply by martynaol** [reply]
Still have the same issue :  "Authorization": "Bearer $S{token}" . I decided to use returns

**Reply by ASaiAnudeep** [reply]
Able to replicate the bug. Creating an issue https://github.com/pactumjs/pactum/issues/137 for the same.

**Reply by jd4u** [reply]
Using the suggested method, the requested header value output is `"Bearer Bearer $S{userAt}"`, userAt is used in stores call.
- The call is made without transforming the $S{userAt}
- The call is made with doubling prefix of $S{userAt} i.e. "Bearer "

[with pactum@3.4.0, node 16.20.0]

Can this be a bug?

**Reply by ASaiAnudeep** [reply]
This happens when no value is stored and `$S{userAt}` returns undefined.

**Reply by jd4u** [reply]
- `$S{userAt}` is not transformed in output... it just emits itself.
- value of access_token is received from request
Code sample is given here ...

**Comment by jd4u** [comment]
Below is the test code...
```js
beforeAll(async () => {

       await pactum
            .spec()
            .post('/auth/signin')
            .withBody(dtoLogin)
            .expectStatus(200)
            .stores('userAt', 'access_token');

        pactum.request.setDefaultHeaders(
            'Authorization',
            `Bearer $S{userAt}`,
        );
        console.log('pactum store done in before');
    });
    describe('Get me', () => {
        it('should get current user', () => {
            return pactum
                .spec()
                .get('/user/me')
                .inspect()
                .expectStatus(200);
        });
    });
```
Login request output is `{access_token:"token", refresh_token:""}`

Below is console log

```stdout
console.warn
    {
      "url": "http://localhost:3344/user/me",
      "method": "GET",
      "path": "/user/me",
      "headers": {
        "Authorization": "Bearer Bearer $S{$userAt}"
      },
      "timeout": 3000
    }

```

**Reply by jd4u** [reply]
- with `beforeAll( async ()=>{}`, the output value repeats the prefix i.e. "Bearer Bearer $S{userAt}"
- with `beforeAll(()=>{}`, the output value DO NOT repeat the prefix i.e. "Bearer $S{userAt}"
- **The stored value output is not happening**
